### PR TITLE
Access to context in tests

### PR DESCRIPTION
**Pull request by sfc-gh-pjob**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
* Added a handle to app context to have access to it in tests. We can't create a separate new context in tests, because plugin command registration is done only in the context of the running command.

### CODE REVIEW

File: src/snowcli/app/cli_app.py [hunk 1]
```diff
@@ -35,6 +37,15 @@
 _commands_registration = CommandsRegistrationWithCallbacks(_api.plugin_config_provider)
 
 
+@dataclass
+class AppContextHolder:
+    # needed to access the context from tests
+    app_context: Optional[Context] = None
```
Comment (sfc-gh-astus) [inline src/snowcli/app/cli_app.py:43]: I'm not sure if we should add something to code just for testing

Comment (sfc-gh-pjob) [reply src/snowcli/app/cli_app.py:43]: That's a good point, but I don't see any other alternative.

Comment (sfc-gh-jsikorski) [reply src/snowcli/app/cli_app.py:43]: The alternative is to add callback option. So maybe we should wait for @sfc-gh-turbaszek decision - which is better?

Comment (sfc-gh-jsikorski) [reply src/snowcli/app/cli_app.py:43]: Yes, it does - we get the list of params and commands, so it`s easy to check, if help is present

Comment (sfc-gh-pjob) [reply src/snowcli/app/cli_app.py:43]: Ok but, am I wrong thinking that the callback is also only for purpose of these tests?

Comment (sfc-gh-jsikorski) [reply src/snowcli/app/cli_app.py:43]: Yes it is

**Review by sfc-gh-jsikorski — APPROVED** [review]

File: src/snowcli/app/cli_app.py (hunk 1 above)
Comment (sfc-gh-pjob) [reply src/snowcli/app/cli_app.py:43]: During our today's meeting we agreed that we accept my solution.

**Pull request merged**